api/scanner/periodic_scanner/periodic_scanner.go

```diff
@@ -1,26 +1,39 @@
 package periodic_scanner
 
 import (
-	"log"
+	"fmt"
 	"sync"
 	"time"
 
 	"github.com/photoview/photoview/api/graphql/models"
+	"github.com/photoview/photoview/api/log"
 	"github.com/photoview/photoview/api/scanner/scanner_queue"
 	"gorm.io/gorm"
 )
 
+type ScannerQueue interface {
+	AddAllToQueue() error
+}
+
+type RealScannerQueue struct{}
+
+func (r *RealScannerQueue) AddAllToQueue() error {
+	return scanner_queue.AddAllToQueue()
+}
+
 type periodicScanner struct {
 	ticker         *time.Ticker
 	ticker_changed chan bool
+	done           chan struct{}
 	mutex          *sync.Mutex
```

Comment:
As refactoring, could you rename this `mutex` as `tickerLocker` and put it together with `ticker`? From the code, this locker only protects `ticker` when the interval is changing.

Also change the type to `sync.Mutex` instead of `*sync.Mutex`.